src/middlewared/middlewared/plugins/vm/vms.py

```diff
@@ -99,10 +99,10 @@ def bootloader_ovmf_choices(self):
     @private
     def extend_context(self, rows, extra):
         status = {}
-        if rows:
+        if rows and self._is_kvm_supported():
```

Comment:
`self._is_kvm_supported()` is called potentially N times + 1 (N being the number of `rows`). Please call this once and store it in a variable.

Reply:
@yocalebo the implementation of this function already has a cached variable so no extra processing is done..should i still save it as a var before?

Reply:
Yes, because it's not immediately obvious that the function returns a cached variable.

Reply:
I also have no idea how that cache value is updated so you could, in theory, have a condition where you call it once and it returns true and then the very next time you call it, it returns false.

Reply:
I am not sure how that would be possible?
I mean we check if kvm is supported on the machine - that won't change on a running machine?

I'll make changes to use the variable but clarifying your last comment